File: examples/5/join.py
#!/usr/local/bin/python3.5 -u

def my_join(separator, list):
    if not list: return ""
    string = list.pop(0)
    for thing in list:
        string += separator + thing
    return string

a = ["Fish Fingers", "and", "Custard"];
b = my_join("   ",  a);
print(b)

File: examples/5/join.pl
#!/usr/bin/perl -w

sub my_join {
    my ($separator, @list) = @_;
    return "" if !@list;
    my $string = shift @list;
    foreach $thing (@list) {
        $string .= $separator . $thing;
    }
    return $string;
}

@a = ("Fish Fingers", "and", "Custard");
$b = my_join("   ",  @a);
print "$b\n"
